src/trade/TradeCard.js

```diff
@@ -169,7 +151,9 @@ export default class TradeCard extends Component {
             serverContractModelToChartContractModel(lastBoughtContract);
 
         // contract error is not tied to trade, but symbol, thus not in tradeErrors
-        const tradeError = (propsContract ? propsContract.error : undefined) || errorToShow(tradeErrors);
+        const tradeError = propsContract ?
+            Object.assign(tradeErrors, { contractError: propsContract.error }) :
```

Comment:
The more I look at the code, the more I am in favor of having arrays, not objects.